source/extensions/filters/http/oauth2/oauth_client.cc

```diff
@@ -40,6 +40,7 @@ void OAuth2ClientImpl::asyncGetAccessToken(const std::string& auth_code,
   const std::string body = fmt::format(GetAccessTokenBodyFormatString, auth_code, encoded_client_id,
                                        encoded_secret, encoded_cb_url);
   request->body().add(body);
+  request->headers().setContentLength(body.length());
```

Comment:
If users do want chunker transfer and not set Content-length, how to deal with it?

Reply:
I thought about that too, but thought that it might be OK to always disable chunked transfer. But I've now looked through the code and tried to add an optional `disable_chunked_transfer` configuration parameter, which defaults to `false` to preserve the current behavior.

Reply:
Sound good to me.